### DISCUSSION TITLE
What do the `flat_ner` and `span_mode` parameters do?

### DISCUSSION THREAD

**Discussion by omri374** [urchade/GLiNER#220] [Q&A]
I'm trying to better understand how these two parameters affect the output. Any intuition would be great, especially around recall vs. precision of entity recognition, i.e. how does choosing these parameters influence the types and spans of entities detected.

**Comment by urchade** [comment]
Hello @omri374 

- **`span_mode`** determines how spans are represented and does not change after training, I don't think it influence much recall/precision
- **`flat_ner`** enables or disables nested entities (set `flat_ner=False` for nested entities).  
- **`multi_label`** allows a single span to have multiple entity types.  

To increase recall, you can set `flat_ner=False` and decrease the `threshold`.

Let me know if you have any further questions!

**Reply by omri374** [reply]
Thanks @urchade, could you give an example of a nested entity compared to a flat one?

**Comment by urchade** [comment]
@omri374 `flat_ner` `(a)` does not allow overlapping entities, while nested `(b)` does (`flat_ner=False`)

![image](https://github.com/user-attachments/assets/66f53160-bb74-4f8a-9970-eced8f73ae0b)

**Reply by omri374** [reply]
Thanks
